Traceback (most recent call last):
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/mmccull/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# cos basis function in theta
def theta_basis(theta,i):
    return np.cos(i*theta)
thetaVals=np.arange(0,np.pi,0.1)
for i in range(5):
    label = "basis function " + str(i)
    plt.plot(thetaVals,theta_basis(thetaVals,i),label=label)
plt.legend();
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[1], line 4[0m
[1;32m      2[0m [38;5;28;01mdef[39;00m [38;5;21mtheta_basis[39m(theta,i):
[1;32m      3[0m     [38;5;28;01mreturn[39;00m np[38;5;241m.[39mcos(i[38;5;241m*[39mtheta)
[0;32m----> 4[0m thetaVals[38;5;241m=[39m[43mnp[49m[38;5;241m.[39marange([38;5;241m0[39m,np[38;5;241m.[39mpi,[38;5;241m0.1[39m)
[1;32m      5[0m [38;5;28;01mfor[39;00m i [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;241m5[39m):
[1;32m      6[0m     label [38;5;241m=[39m [38;5;124m"[39m[38;5;124mbasis function [39m[38;5;124m"[39m [38;5;241m+[39m [38;5;28mstr[39m(i)

[0;31mNameError[0m: name 'np' is not defined

